crates/ethereum/payload/src/lib.rs

```diff
@@ -314,6 +314,7 @@ where
     }
 
     // calculate the requests and the requests root
+    // TODO (garwah): Figure out how to replace this check now that chain spec has been removed.
```

Comment:
@mattsse It looks like `chainspec` is used heavily to determine what hardforks are active here, or threaded down eg. https://github.com/paradigmxyz/reth/blob/afac7e04af1534604bac6c76af71f2e886c605ae/crates/evm/src/system_calls/eip4788.rs#L74-L76

Do we have an alternative here for these checks or could we pull a chainspec from somewhere else?

Reply:
Alternatively, we move the `EthereumHardForks` helper methods off ChainSpec onto something else